modules/internal/String.chpl

```diff
@@ -617,11 +643,38 @@ module String {
       }
     }
 
+    /*
+      :returns: The value of the `i` th byte as an integer.
+    */
+    proc byte(i: int): uint(8) {
+      return this[i].buff[0];
```

Comment:
Also creates a string temporary to access a single byte. I can see why that might help multilocale, but it seems at least worth an issue.

Reply:
Maybe it could be like
``` chapel
var localThis: string = this.localize();
return localThis.buff[i];
```

?

Reply:
Now better for multilocale, thanks.